### PR TITLE
Use paradigm filler in cree_inflection_generator

### PR DESCRIPTION
**Pull request by eddieantonio**
We used to have duplicated code with regards to "expanding" analyses given an existing analysis. In particular, the `cree_inflection_generator` module that is used in the dictionary importer was not updated to use the updated paradigm filler.

This PR did a few refactorings, added a few tests, and uses the default paradigm filler in the `cree_inflection_generator()`.

@Madoshakalaka: how does this PR affect your JIT branch?

### CODE REVIEW

File: CreeDictionary/DatabaseManager/cree_inflection_generator.py [hunk 1]
```diff
@@ -46,41 +23,39 @@ def expand_inflections(
...
         lemma_category = fst_analysis_parser.extract_lemma_text_and_word_class(analysis)
         assert lemma_category is not None
         lemma, category = lemma_category
-        if category.is_verb() or category.is_noun():
-            generated_analyses = [
-                generated_analysis.replace("{{ lemma }}", lemma)
-                for generated_analysis in flattened_layouts[category]
-            ]
+
+        if category.has_inflections():
```
Comment (eddieantonio) [inline CreeDictionary/DatabaseManager/cree_inflection_generator.py:38]: This was a bug waiting to happen! So it turns out that some word classes apart from nouns and verbs _might_ have inflections? I think the pronouns and diectic words have inflections, maybe.

File: CreeDictionary/DatabaseManager/cree_inflection_generator.py [hunk 2]
```diff
@@ -46,41 +23,39 @@ def expand_inflections(
...
+            generated_analyses = list(paradigm_filler.expand_analyses(lemma, category))
         else:
             generated_analyses = [analysis]
+
         to_generated[analysis] = generated_analyses
-        fat_generated_analyses.extend(generated_analyses)
```
Comment (eddieantonio) [inline CreeDictionary/DatabaseManager/cree_inflection_generator.py:71]: I liked the name "fat_generated_analyses", but I imagine it more as a queue of analyses ready to be inflected by the generator FST.

File: CreeDictionary/tests/utils_tests/data/layouts/noun-nid-full.layout [hunk 3]
```diff
@@ -24,7 +24,7 @@ tooltips:
 | "someone's"               | =N+I+D+PxX+Sg            | =N+I+D+PxX+Pl           | =N+I+D+PxX+Loc             |
 | "my"                      | =N+I+D+Px1Sg+Sg          | =N+I+D+Px1Sg+Pl         | =N+I+D+Px1Sg+Loc           |
 | "your (one)"              | =N+I+D+Px2Sg+Sg          | =N+I+D+Px2Sg+Pl         | =N+I+D+Px2Sg+Loc           |
-| "his/her"                 |                          |                         | =N+I+D+Px3Sg+Loc           |
```
Comment (eddieantonio) [inline CreeDictionary/tests/utils_tests/data/layouts/noun-nid-full.layout:27]: I think these forms were just forgotten in a previous iteration?

File: CreeDictionary/utils/paradigm_filler.py [hunk 4]
```diff
@@ -118,3 +119,50 @@ def fill_paradigm(
...
+        """
+        analyses = self.expand_analyses(lemma, wordclass)
+        return self._generator.feed_in_bulk_fast(analyses)
+
+    def inflect_all(self, lemma: str, wordclass: WordClass) -> Set[str]:
+        """
+        Return a set of all inflections for a particular lemma.
+        """
+        all_inflections = self.inflect_all_with_analyses(lemma, wordclass).values()
+        return set(form for word in all_inflections for form in word)
```
Comment (eddieantonio) [inline CreeDictionary/utils/paradigm_filler.py:139]: Honestly, I'm not sure if these methods in and of themselves are useful, BUT, they certainly are easier and clearer to test than `expand_analyses()` which they make use of. In particular, `inflect_all()` is straightforward to test, and makes use of `expand_analyses()` (albeit transitively).

File: CreeDictionary/DatabaseManager/cree_inflection_generator.py (hunk 1 above)
Comment (Madoshakalaka) [reply CreeDictionary/DatabaseManager/cree_inflection_generator.py:38]: cool!

**Pull request merged**